android/tangram/src/com/mapzen/tangram/MapData.java

```diff
@@ -0,0 +1,102 @@
+package com.mapzen.tangram;
+
+import java.util.List;
+
+public class MapData {
+
+    /**
+     * Construct a new MapData object for adding drawable data to the map
+     * @param name Name of the data source in the scene file for styling this object's data
+     */
+    public MapData(String name) {
+        this.name = name;
```

Comment:
This is a pretty common Java pattern, see:
https://github.com/mapzen/eraser-map/blob/master/app/src/main/java/com/mapzen/erasermap/AndroidModule.java#L25,
https://github.com/mapzen/eraser-map/blob/master/app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt#L53

On Tue, Sep 8, 2015 at 6:32 PM, Varun <notifications@github.com> wrote:

> In android/tangram/src/com/mapzen/tangram/MapData.java
> <https://github.com/tangrams/tangram-es/pull/267#discussion_r38988512>:
>
> > @@ -0,0 +1,102 @@
> > +package com.mapzen.tangram;
> > +
> > +import java.util.List;
> > +
> > +public class MapData {
> > +
> > +    /**
> > +     * Construct a new MapData object for adding drawable data to the map
> > +     * @param name Name of the data source in the scene file for styling this object's data
> > +     */
> > +    public MapData(String name) {
> > +        this.name = name;
>
> Can we remove the use of this here, unless its a Java way of doing
> things.
> To me since id and name are private class members its implicit, and rename
> the argument from name to something else.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/267/files#r38988512>.
>